airbyte-integrations/connectors/source-bing-ads/unit_tests/integrations/test_report_stream.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+
+from pathlib import Path
+from typing import Any, Dict, Optional
+from unittest import TestCase
+from unittest.mock import patch
+
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.test.entrypoint_wrapper import EntrypointOutput, read
+from airbyte_cdk.test.mock_http import HttpMocker
+from bingads.v13.reporting.reporting_service_manager import ReportingServiceManager
+from catalog_builder import BingAdsCatalogBuilder
+from client_builder import build_request, response_with_status
+from config import ConfigBuilder
+from source_bing_ads.client import Client
+from source_bing_ads.source import SourceBingAds
+from suds.transport.https import HttpAuthenticated
+from suds_response_mock import mock_http_authenticated_send
+
+
+class TestReportStream(TestCase):
+    @property
+    def _config(self):
+        return ConfigBuilder().build()
+
+    def _state(self, file: str) -> Path:
+        Path(__file__).parent.parent / f"resource/state/{file}.json"
+
+    def _download_file(self, file: Optional[str] = None) -> Path:
+        """
+        Returns path to temporary file of downloaded data that will be use in read.
+        Base file should be named as {file_name}.cvs in resource/response folder.
+        """
+        if file:
+            path_to_tmp_file = Path(__file__).parent.parent / f"resource/response/{file}.csv"
+            return path_to_tmp_file
+        return Path(__file__).parent.parent / "resource/response/non-existing-file.csv"
+
+    def _get_client(self, http_mocker, config: dict) -> Client:
+        http_mocker.post(
+            request=build_request(self._config),
+            responses=response_with_status("oauth", 200)
+        )
+        client = Client(**config)
+        return client
+
+    def read_stream(
+            self, stream_name: str, sync_mode: SyncMode, config: Dict[str, Any], pk: list[str], stream_data_file: str = None,
+            state: Optional[Dict[str, Any]] = None, expecting_exception: bool = False,
+    ) -> EntrypointOutput:
+        with patch.object(HttpAuthenticated, "send", mock_http_authenticated_send):
+            with patch.object(ReportingServiceManager, "download_file", return_value=self._download_file(stream_data_file)):
+                catalog = BingAdsCatalogBuilder().with_stream(stream_name, sync_mode, pk).build()
+                return read(SourceBingAds(), config, catalog, state, expecting_exception)
+
+
+class TestSuiteReportStream(TestReportStream):
+    stream_name: str = False
```

Comment:
Why do we have a default value with type `bool` for the `str` variable?

Reply:
updated with default None, we use stream name to identify that this is a inherited test class and we should run test suite.